system/jlib/jevent.cpp

```diff
@@ -1757,6 +1821,513 @@ IEventVisitor* createDumpCSVEventVisitor(IBufferedSerialOutputStream& out)
+            }
+            return true;
+        }
+        bool matches(EventType evt, __uint64 value) const override
+        {
+            if (MetaFileInformation == evt)
+            {
+                // Cache the presumed EvAttrFileId value for use when matching EvAttrPath.
+                curMetaId = value;
+                return true;
+            }
+            return matchedPaths.count(value) ||  UnsignedFilterTerm::matches(evt, value);
+        }
+        bool acceptToken(const char* token) override
+        {
+            if (UnsignedFilterTerm::acceptToken(token))
+                return true;
+            // Token not a number or range; assume it is a path pattern
+            return helper.acceptToken(token);
+        }
+
+        StringMatchHelper helper;
+        mutable std::set<__uint64> matchedPaths;
+        mutable __uint64 curMetaId{0};
+    };
+
+    struct TimestampFilterTerm : public UnsignedFilterTerm
+    {
+        bool acceptToken(const char* token) override
+        {
+            // Digits and hyphen only is presumed to be a nanosecond timestamp range.
+            std::string tmp(token);
+            if (tmp.find_first_not_of("0123456789-") == std::string::npos)
+                return UnsignedFilterTerm::acceptToken(token);
+
+            // All else is presumed to be a date/time string or a date/time range.
+            __uint64 first = 0, last = UINT64_MAX;
+            CDateTime dt;
+            const char* delim = nullptr;
+            if (*token == '-') // range starting at 0
+            {
+                dt.setString(token + 1, &delim);
+                if (delim && *delim) // bad content
+                    return false;
+                last = dt.getTimeStampNs();
+            }
+            else // single value or range not starting at 0
+            {
+                dt.setString(token, &delim);
+                first = dt.getTimeStampNs();
+                if (!delim) // single value
+                    last = first;
+                else
+                {
+                    trim(delim);
+                    if ('-' == *delim) // a range...
+                    {
+                        token = delim + 1;
+                        if (*token) // with bounded end
+                        {
+                            trim(token);
+                            delim = nullptr;
+                            dt.setString(token, &delim);
+                            if (delim && *delim) // bad content
+                                return false;
+                            last = dt.getTimeStampNs();
+                        }
+                        else // with unbounded end
+                            last = first;
+                    }
+                    else // bad content
+                        return false;
+                }
+            }
+            return acceptRange(first, last);
+        }
+
+        void trim(const char*& token)
```

Comment:
better named as skipSpaces - because trim generally means removed them from the end.  Can be a static/non member function.